hedera-node/hedera-app/src/main/java/com/hedera/node/app/signature/impl/SignatureVerificationFutureImpl.java

```diff
@@ -0,0 +1,192 @@
+     * very nice.
+     *
+     * <p>instead, we can just do a "best effort" cancel, which is what we will do. If the future exists, we will
+     * cancel it. If it does not exist, we will let it be.
+     */
+    @Override
+    public boolean cancel(final boolean mayInterruptIfRunning) {
+        // If we're already done, then we can't cancel again
+        if (isDone()) {
+            return false;
+        }
+
+        // Try to cancel the underlying future, if it exists. If the underlying Future doesn't exist, then we just skip
+        // it. Best effort.
+        canceled = true;
+        final var future = txSig.getFuture();
+        if (future != null) {
+            return future.cancel(mayInterruptIfRunning);
+        }
+
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isCancelled() {
+        return canceled;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Due to the asynchronous nature of {@link TransactionSignature}, it is possible that at the time this method
+     * is called, one or more {@link TransactionSignature}s do not yet have a {@link Future}. If that is the case,
+     * then we are clearly not done!
+     */
+    @Override
+    public boolean isDone() {
+        if (canceled) {
+            return true;
+        }
+
+        final var future = txSig.getFuture();
+        return future != null && future.isDone();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Blocks on each {@link Future}. Due to the asynchronous nature of {@link TransactionSignature}, it is possible
+     * that a {@link Future} has not been assigned to one or more {@link TransactionSignature}s. In that case, we will
+     * wait for the {@link Future} before proceeding.
+     */
+    @NonNull
+    @Override
+    public SignatureVerification get() throws InterruptedException, ExecutionException {
+        txSig.waitForFuture().get(); // Wait for the future to be assigned and completed
+        return new SignatureVerificationImpl(key, hollowAccount, txSig.getSignatureStatus() == VALID);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Blocks on each {@link Future}. Due to the asynchronous nature of {@link TransactionSignature}, it is possible
+     * that a {@link Future} has not been assigned to one or more {@link TransactionSignature}s. In that case, we will
+     * wait for the {@link Future} before proceeding (up to the timeout).
+     */
+    @NonNull
+    @Override
+    public SignatureVerification get(long timeout, TimeUnit unit)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        var time = System.currentTimeMillis();
+        var millisRemaining = unit.toMillis(timeout);
+        while (txSig.getFuture() == null) {
+            final var now = System.currentTimeMillis();
+            millisRemaining -= (now - time);
+            if (millisRemaining <= 0) {
+                // If there was no time left, then TimeoutException
```

Comment:
Possibly cleaner rewrite:
```
        final long deadline = System.currentTimeMillis() + unit.toMillis(timeout);
        while (txSig.getFuture() == null) {
            if (System.currentTimeMillis() > deadline) {

```
Further down
```
        txSig.getFuture().get(deadline - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
```

Reply:
I love it. Will integrate and push.